File: leet/0028-implement-strStr/implement-strStr.go
func strStr(haystack string, needle string) int {
    if len(needle) == 0 {return 0}
    if len(haystack) == 0 {return -1}
    
    for i := 0; i < len(haystack); i++ {
        allSame := true
        for j := 0; j < len(needle); j++ {
            if(i + j >= len(haystack)){
                return -1
            } else if(haystack[i + j] != needle[j]){
                allSame = false
                break;
            }
        }
        
        if(allSame) {return i}
    }
    
    return -1
}

File: leet/0028-implement-strStr/implement-strStr.py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        if needle == "" : return 0
        
        for i in range(len(haystack)):            
            if haystack[i] == needle[0] and len(haystack) - i >= len(needle):
                hs, hn = i, 0
                
                while hn < len(needle) and haystack[hs] == needle[hn]:
                    hs += 1; hn += 1
                    
                if hn == len(needle):
                    return i
                
        return -1
